src/payment/strategies/braintree/braintree-credit-card-payment-strategy.ts

```diff
@@ -84,37 +104,93 @@ export default class BraintreeCreditCardPaymentStrategy implements PaymentStrate
         throw error;
     }
 
-    private _isUsingVaulting(paymentData: PaymentInstrument): boolean {
-        if (isCreditCardLike(paymentData)) {
-            return Boolean(paymentData.shouldSaveInstrument);
+    private async _preparePaymentData(payment: OrderPaymentRequestBody, billingAddress: Address, orderAmount: number): Promise<PaymentInstrument & PaymentInstrumentMeta> {
+        const commonPaymentData = { deviceSessionId: this._deviceSessionId };
+
+        if (this._isHostedFormInitialized) {
```

Comment:
this method seems a bit long, do you think it might be worth splitting into different methods, so that there's only one nesting level?
```js
if (this._isHostedFormInitialized) {
  return this._prepareHostedFormPaymentData(...)
}

// ...

Reply:
Yes, good suggestion 👍